mteb/tasks/BitextMining/multilingual/BibleNLPBitextMining.py

```diff
@@ -0,0 +1,955 @@
+    "yka_Latn",  # Yakan
+    "yle_Latn",  # Yele
+    "yml_Latn",  # Iamalele
+    "yon_Latn",  # Yongkom
+    "yor_Latn",  # Yoruba, primarily uses the Latin script
+    "yrb_Latn",  # Yareba
+    "yre_Latn",  # Yaouré
+    "yss_Latn",  # Yessan-Mayo
+    "yuj_Latn",  # Karkar-Yuri
+    "yut_Latn",  # Yopno
+    "yuw_Latn",  # Yau (Morobe Province)
+    "yva_Latn",  # Yawa
+    "zaa_Latn",  # Sierra de Juárez Zapotec
+    "zab_Latn",  # San Juan Guelavía Zapotec
+    "zac_Latn",  # Ocotlán Zapotec
+    "zad_Latn",  # Cajonos Zapotec
+    "zai_Latn",  # Isthmus Zapotec
+    "zaj_Latn",  # Zaramo
+    "zam_Latn",  # Miahuatlán Zapotec
+    "zao_Latn",  # Ozolotepec Zapotec
+    "zap_Latn",  # Zapotec, uses Latin script across many varieties
+    "zar_Latn",  # Rincón Zapotec
+    "zas_Latn",  # Santo Domingo Albarradas Zapotec
+    "zat_Latn",  # Tabaa Zapotec
+    "zav_Latn",  # Yatzachi Zapotec
+    "zaw_Latn",  # Mitla Zapotec
+    "zca_Latn",  # Coatecas Altas Zapotec
+    "zga_Latn",  # Ganza
+    "zia_Latn",  # Zia
+    "ziw_Latn",  # Zigula
+    "zlm_Latn",  # Malay (individual language), Latin script
+    "zos_Latn",  # Francisco León Zoque
+    "zpc_Latn",  # Choapan Zapotec
+    "zpl_Latn",  # Lachixío Zapotec
+    "zpm_Latn",  # Mixtepec Zapotec
+    "zpo_Latn",  # Amatlán Zapotec
+    "zpq_Latn",  # Zoogocho Zapotec
+    "zpu_Latn",  # Yalálag Zapotec
+    "zpv_Latn",  # Chichicapan Zapotec
+    "zpz_Latn",  # Texmelucan Zapotec
+    "zsr_Latn",  # Southern Rincon Zapotec
+    "ztq_Latn",  # Quioquitani-Quierí Zapotec
+    "zty_Latn",  # Yatee Zapotec
+    "zyp_Latn",  # Zyphe Chin
+]
+
+_SPLIT = ["train"]
+
+_N = 256
+
+
+def extend_lang_pairs_english_centric() -> dict[str, list[str]]:
+    # add all language pairs with english as source or target
+    hf_lang_subset2isolang = {}
+    for x in _LANGUAGES:
+        if "-" not in x:
+            for y in _LANGUAGES:
+                if x != y and "eng_Latn" in [x, y]:
+                    pair = f"{x}-{y}"
+                    hf_lang_subset2isolang[pair] = [
+                        x.replace("_", "-"),
+                        y.replace("_", "-"),
+                    ]
+    return hf_lang_subset2isolang
+
+
+_LANGUAGES_MAPPING = extend_lang_pairs_english_centric()
+
+
+class BibleNLPBitextMining(AbsTaskBitextMining, CrosslingualTask):
+    metadata = TaskMetadata(
+        name="BibleNLPBitextMining",
+        dataset={
+            "path": "davidstap/biblenlp-corpus-mmteb",
+            "revision": "264a18480c529d9e922483839b4b9758e690b762",
+            "split": f"train[:{_N}]",
+            "trust_remote_code": True,
+        },
+        description="Partial Bible translations in 829 languages, aligned by verse.",
+        reference="https://arxiv.org/abs/2304.09919",
+        type="BitextMining",
+        category="s2s",
+        eval_splits=_SPLIT,
+        eval_langs=_LANGUAGES_MAPPING,
+        main_score="f1",
+        date=("2023-04-09", "2023-04-09"),
```

Comment:
Here is an interesting one, cause [for consistency](https://github.com/embeddings-benchmark/mteb/pull/545#discussion_r1577555935) it should be the date of when the source was written 🤔 or maybe in this case a period when the translations where made if such information available? One way or another it requires some adnotation in the comment.

Reply:
I see, interesting indeed. For English, I mostly used the World English Bible (WEB) versions. On Wikipedia I can find: first draft in 1997, completed in 2020, so I will add this information. 

It seems impossible, or at the minimum infeasible, to add information for all other languages.